src/Matrix.jl

```diff
@@ -3686,6 +3686,33 @@ function nullspace(M::MatElem{T}) where {T <: FieldElement}
    return nullity, X
 end
 
+###############################################################################
+#
+#   Nilpotency
+#
+###############################################################################
+
+@doc raw"""
+    is_nilpotent(A::MatrixElem{T}) where {T <: RingElement}
+
+Return if `A` is nilpotent, i.e. if there exists a natural number $k$
+such that $A^k = 0$. If `A` is not square an exception is raised.
+"""
+function is_nilpotent(A::MatrixElem{T}) where {T <: RingElement}
+  !is_square(A) && error("Dimensions don't match in is_nilpotent")
+  is_zero(tr(A)) || return false
+  n = nrows(A)
+  A = deepcopy(A)
+  i = 1
+  is_zero(A) && return true
+  while i < n
+    i *= 2
+    A = mul!(A, A, A)
```

Comment:
It might or might not be faster to compute the minimal or characteristic polynomial (if there are dedicated optimized methods for this and the given matrix type

Reply:
yeah that's true. but for that one would need to know which specialized methods exist in which cases, which I honestly don't